### ISSUE TITLE
1.21 cluster fails to provision ilb load balancer with "cannot find public IP with IP address 10.1.0.6"

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes-sigs/cluster-api-provider-azure#1347]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I ran `./scripts/ci-e2e.sh` 

```
[1] • Failure [1007.159 seconds]                                                                                   
[1] Workload cluster creation                                                                                      
[1] /home/jstur/projects/cluster-api-provider-azure/test/e2e/azure_test.go:38                                      
[1]   Creating a Windows Enabled cluster                                                                           
[1]   /home/jstur/projects/cluster-api-provider-azure/test/e2e/azure_test.go:447                                   
[1]     With 3 control-plane nodes and 1 Linux worker node and 1 Windows worker node [It]                                                                                                 
[1]     /home/jstur/projects/cluster-api-provider-azure/test/e2e/azure_test.go:449                                 
[1]                                                                                                                
[1]     Timed out after 300.028s.                                                                                  
[1]     Service default/web-windows-ilb failed    
```

Service:

```
I0428 21:01:37.044134       1 deployment_controller.go:490] "Error syncing deployment" deployment="default/web" err="Operation cannot be fulfilled on deployments.apps \"web\": the object has been modified; please apply your changes to the latest version and try again"
W0428 21:02:19.121996       1 azure_standard.go:365] serviceOwnsFrontendIP: unexpected error when finding match public IP of the service web-ilb with loadBalancerLP 10.1.0.6: findMatchedPIPByLoadBalancerIP: cannot find public IP with IP address 10.1.0.6 in resource group capz-e2e-2jlzhw
W0428 21:02:19.163921       1 azure_standard.go:365] serviceOwnsFrontendIP: unexpected error when finding match public IP of the service web-ilb with loadBalancerLP 10.1.0.6: findMatchedPIPByLoadBalancerIP: cannot find public IP with IP address 10.1.0.6 in resource group capz-e2e-2jlzhw
W0428 21:02:19.195953       1 azure_standard.go:365] serviceOwnsFrontendIP: unexpected error when finding match public IP of the service web-ilb with loadBalancerLP 10.1.0.6: findMatchedPIPByLoadBalancerIP: cannot find public IP with IP address 10.1.0.6 in resource group capz-e2e-2jlzhw
W0428 21:03:05.609996       1 azure_utils.go:129] parseTags: error when parsing key-value pair , would ignore this one
I0428 21:03:48.227155       1 deployment_controller.go:490] "Error syncing deployment" deployment="default/web-windows" err="Operation cannot be fulfilled on deployments.apps \"web-windows\": the object has been modified; please apply your changes to the latest version and try again"
I0428 21:03:48.255605       1 deployment_controller.go:490] "Error syncing deployment" deployment="default/web-windows" err="Operation cannot be fulfilled on deployments.apps \"web-windows\": the object has been modified; please apply your changes to the latest version and try again"
E0428 21:05:08.596855       1 controller.go:307] error processing service default/web-windows-ilb (will retry): failed to ensure load balancer: not a vmss instance
I0428 21:05:08.597006       1 event.go:291] "Event occurred" object="default/web-windows-ilb" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message="Error syncing load balancer: failed to ensure load balancer: not a vmss instance"
E0428 21:05:13.654134       1 controller.go:307] error processing service default/web-windows-ilb (will retry): failed to ensure load balancer: not a vmss instance
I0428 21:05:13.654376       1 event.go:291] "Event occurred" object="default/web-windows-ilb" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message="Error syncing load balancer: failed to ensure load balancer: not a vmss instance"
E0428 21:05:23.709396       1 controller.go:307] error processing service default/web-windows-ilb (will retry): failed to ensure load balancer: not a vmss instance
I0428 21:05:23.709442       1 event.go:291] "Event occurred" object="default/web-windows-ilb" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message="Error syncing load balancer: failed to ensure load balancer: not a vmss instance"
```

**What did you expect to happen:**
Service LB to be deployed properly

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: head
- Kubernetes version: (use `kubectl version`): 1.21
- OS (e.g. from `/etc/os-release`): 

```
capz-e2e-2jlzhw-admin@capz-e2e-2jlzhw|kube-system]  ~/projects/cluster-api-provider-azure   master ±  k get nodes  
NAME                                  STATUS   ROLES                  AGE   VERSION
capz-e2e-2jlzhw-control-plane-2ctwx   Ready    control-plane,master   39m   v1.21.0
capz-e2e-2jlzhw-control-plane-j7hjm   Ready    control-plane,master   41m   v1.21.0
capz-e2e-2jlzhw-control-plane-qrh8h   Ready    control-plane,master   43m   v1.21.0
capz-e2e-2jlzhw-md-0-z7s4s            Ready    <none>                 42m   v1.21.0
capz-e2e-r7lbz                        Ready    <none>                 40m   v1.21.0
```

**Comment by CecileRobertMichon** [comment]
@feiskyer @nilo19 we're observing the above failure with in-tree but not out-of-tree. I suspect it might be caused by  https://github.com/kubernetes/kubernetes/pull/94991

**Comment by nilo19** [comment]
What is the sepc of the testing service `web-windows-ilb`?

**Comment by jsturtevant** [comment]
@nilo19 it is generated here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a92161fa208a2cd46dd1eac9731d03df1abe1170/test/e2e/kubernetes/deployment/deployment.go#L144

**Comment by nilo19** [comment]
@jsturtevant could you show me the full log from `Ensuring load balancer for service xxx` to `Ensured load balancer`?

**Comment by nilo19** [comment]
The warning message is expected because we perform 2 reconcile loops for each service. For internal services, the first round is reconciling the internal lb and the second round is reconciling the external lb. This warning message comes from the second round. I still need detailed logs to figure out where did the error message `not a vmss instance` come from.

**Comment by nilo19** [comment]
https://github.com/kubernetes/kubernetes/pull/101752 to fix this one

**Comment by jsturtevant** [comment]
should be resolved with #706

kubernetes/kubernetes#104066
kubernetes/kubernetes#104068
kubernetes/kubernetes#104029
